### PR TITLE
Throw error when an E2 UDF doesn't return a value

### LINKED ISSUE #519: [E2] Lua error on assigning nil arrays/tables to variables
Assigning nil arrays to variables cause a lua error and make the E2 chip error. The error appears on gmod_wire_expression2/core/array.lua:104.

This can happen when a UDF that should return an array doesn't do it, here's an example:

```
function array test() {}
Array = test()
```

The bug can possibly be fixed in many ways, like
1) just assigning a blank array (return {}) if a nil array was specified, which would in my opinion be wrong because forgetting to return a value ending up as a blank array would cause confusion.
2) if nil array specified, return nil
3) or even make functions error if a value is not returned, but this way could possibly break some E2s.

I think that the second way is the best because the user could check if an array was nil or not with a basic "if" check, example:

```
function array test() {}
Array = test()
if ( Array ) {
    print( "test() returned a valid array" )
} else {
    print( "test() returned a nil array" )
}
```

I'm pretty sure that the example would work correctly without any edits to how "if" works, because this example seems to work:

```
#function array test() {return array()}
function array test() {}
if ( test() ) {
    print( "test() returned a valid array" )
} else {
    print( "test() returned a nil array" )
}
```

There's also a similar problem with tables, but with only a quick peek at the code I'm not really sure how that could be fixed.

### PR DESCRIPTION
**Pull request by AbigailBuccaneer**
Fixes #519.

### CODE REVIEW

**Comment by AbigailBuccaneer** [comment]
The only really meaningful diff here is [for the last commit](https://github.com/wiremod/wire/pull/1277/commits/b9a5a8596bcd91595e726addb176d1bcbbf5424f).

File: lua/entities/gmod_wire_expression2/core/functions.lua [hunk 1]
```diff
@@ -27,39 +32,24 @@ local function Function(A,S,Scopes)
...
-
-__e2setcost(20)
-
-registerOperator("function", "", "", function(self, args)
-
-	local Stmt, args = args[2], args[3]
-	local Sig, Return, Args = args[3], args[4], args[6]
-
-	self.funcs[Sig] = Function(Args,Stmt)
+		if Return ~= "" then
```
Comment (bigdogmat) [inline lua/entities/gmod_wire_expression2/core/functions.lua]: Shouldn't this be `ReturnType`?

Comment (AbigailBuccaneer) [inline lua/entities/gmod_wire_expression2/core/functions.lua]: It should! Thank you.

**Comment by Nebual** [comment]
Seems to work. Would be better at compile-time of course (a UDF lacking a return statement when it has a return type in its signature doesn't seem like it'd be hard to catch), but this is something.

**Pull request merged**